picamera2/encoders/h264_encoder.py

```diff
@@ -26,18 +28,24 @@ def __init__(self, bitrate=None, repeat=True, iperiod=None, framerate=None, enab
         :type iperiod: int, optional
         :param framerate: record a framerate in the stream (whether true or not)
         :type framerate: float, optional
+        :param qp: Fixed quantiser from 1 to 51 (disables constant bitrate), default None
+        :type qp: int
         """
         super().__init__(bitrate, V4L2_PIX_FMT_H264)
-        if iperiod is not None:
-            self._controls += [(V4L2_CID_MPEG_VIDEO_H264_I_PERIOD, iperiod)]
-        if repeat:
-            self._controls += [(V4L2_CID_MPEG_VIDEO_REPEAT_SEQ_HEADER, 1)]
+        self.iperiod = iperiod
+        self.repeat = repeat
+        self.qp = qp
         # The framerate can be reported in the sequence headers if enable_sps_framerate is set,
         # but there's no guarantee that frames will be delivered to the codec at that rate!
         self._framerate = framerate
         self._enable_framerate = enable_sps_framerate
 
     def _start(self):
+        if self.iperiod is not None:
+            self._controls += [(V4L2_CID_MPEG_VIDEO_H264_I_PERIOD, self.iperiod)]
```

Comment:
Just thinking because you've moved this code from __init__ to _start, won't self._controls, be concatenated to each time you start the encoder?

Reply:
Indeed, and I've got it wrong for all the controls in there! I think I was wanting folks to be able to set or change these things after initialising the encoder object, so long as they do it before starting (or re-starting after stopping) it. Maybe putting `self._controls = []` at the top would do, or might we need anything more sophisticated?